src/components/text-input/index.md.njk

```diff
@@ -61,17 +61,29 @@ Fluid width inputs will resize with the viewport.
 
 ### Numbers
 
-If it’s likely that the user will need to enter a number and you want to bring up the numeric keypad on a mobile device, set the `inputmode` attribute to `numeric` and the `pattern` attribute to `[0-9]*`. See how to do this in the HTML and Nunjucks tabs in the following example.
+#### Asking for whole numbers
 
-{{ example({group: "components", item: "text-input", example: "number-input", html: true, nunjucks: true, open: false, size: "m"}) }}
+If you're asking the user to enter a whole number and you want to bring up the numeric keypad on a mobile device, set the `inputmode` attribute to `numeric` and the `pattern` attribute to `[0-9]*`. See how to do this in the HTML and Nunjucks tabs in the following example.
 
-Do not use `<input type=”number”>` unless your user research shows that there’s a need for it. With `<input type=”number”>` there’s a risk of users accidentally incrementing a number when they’re trying to do something else - for example, scroll up or down the page. And if the user tries to enter something that’s not a number, there’s no explicit feedback about what they’re doing wrong.
+{{ example({group: "components", item: "text-input", example: "number-input", html: true, nunjucks: true, open: false, size: "m"}) }}
 
 There are specific patterns for:
 
 - [dates](/patterns/dates/)
 - [telephone numbers](/patterns/telephone-numbers/)
 
+#### Asking for numbers with decimal places
+
+If you're asking the user to enter a number that might include a decimal place, use `input type="text"` without `inputmode` or `pattern` attributes.
+
+Do not set the `inputmode` attribute to `decimal` as it causes some devices to bring up a keypad without a period key, preventing users from entering a decimal point.
+
+{{ example({group: "components", item: "text-input", example: "decimal-input", html: true, nunjucks: true, open: false, size: "m"}) }}
+
+#### Avoid using inputs with a type of number
+
+Do not use `<input type=”number”>` unless your user research shows that there’s a need for it. With `<input type=”number”>` there’s a risk of users accidentally incrementing a number when they’re trying to do something else - for example, scroll up or down the page. And if the user tries to enter something that’s not a number, there’s no explicit feedback about what they’re doing wrong.
```

Comment:
Non-blocking suggestion for: 
> And if the user tries to enter something that’s not a number, there’s no explicit feedback about what they’re doing wrong.

It would be good to mention that users are prevented from entering characters. Perhaps something like:
> Some browsers also prevent users from entering non-numeric characters in `<input type=”number”>`, with no explicit feedback to users about what they're doing wrong.

Reply:
This hasn't changed, just moved – so if it's OK with you I'd prefer to address this in a separate PR.